cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_regression_hist_dpc.cpp

```diff
@@ -0,0 +1,185 @@
+
+    const int64_t row_count    = data.get_row_count();
+    const int64_t column_count = data.get_column_count();
+
+    auto arr_data  = row_accessor<const Float>{ data }.pull(queue);
+    auto arr_label = row_accessor<const Float>{ labels }.pull(queue);
+
+    const auto daal_data =
+        interop::convert_to_daal_sycl_homogen_table(queue, arr_data, row_count, column_count);
+    const auto daal_labels =
+        interop::convert_to_daal_sycl_homogen_table(queue, arr_label, row_count, 1);
+
+    /* init param for daal kernel */
+    auto daal_input = reg::training::Input();
+    daal_input.set(reg::training::data, daal_data);
+    daal_input.set(reg::training::dependentVariable, daal_labels);
+
+    auto daal_parameter                        = reg::training::Parameter();
+    daal_parameter.nTrees                      = desc.get_tree_count();
+    daal_parameter.observationsPerTreeFraction = desc.get_observations_per_tree_fraction();
+    daal_parameter.featuresPerNode             = desc.get_features_per_node();
+    daal_parameter.maxTreeDepth                = desc.get_max_tree_depth();
+    daal_parameter.minObservationsInLeafNode   = desc.get_min_observations_in_leaf_node();
+    // TODO take engines from desc
+    daal_parameter.engine            = daal::algorithms::engines::mt2203::Batch<>::create();
+    daal_parameter.impurityThreshold = desc.get_impurity_threshold();
+    daal_parameter.memorySavingMode  = desc.get_memory_saving_mode();
+    daal_parameter.bootstrap         = desc.get_bootstrap();
+    daal_parameter.minObservationsInSplitNode     = desc.get_min_observations_in_split_node();
+    daal_parameter.minWeightFractionInLeafNode    = desc.get_min_weight_fraction_in_leaf_node();
+    daal_parameter.minImpurityDecreaseInSplitNode = desc.get_min_impurity_decrease_in_split_node();
+    daal_parameter.maxLeafNodes                   = desc.get_max_leaf_nodes();
+
+    daal_parameter.resultsToCompute = desc.get_train_results_to_compute();
+
+    auto vimp = desc.get_variable_importance_mode();
+
+    daal_parameter.varImportance = df_interop::convert_to_daal_variable_importance_mode(vimp);
+
+    train_result<Task> res;
+
+    auto daal_result = reg::training::Result();
+
+    /* init daal result's objects */
+    array<Float> arr_oob_err;
+    if (desc.get_train_results_to_compute() &
+        static_cast<std::uint64_t>(train_result_to_compute::compute_out_of_bag_error)) {
+        arr_oob_err = array<Float>::empty(1 * 1);
+        //res.set_oob_err(homogen_table_builder{}.reset(arr_oob_err, 1, 1).build());
+
+        const auto res_oob_err =
+            interop::convert_to_daal_sycl_homogen_table(queue, arr_oob_err, 1, 1);
+        daal_result.set(reg::training::outOfBagError, res_oob_err);
+    }
+
+    array<Float> arr_oob_per_obs_err;
+    if (desc.get_train_results_to_compute() &
+        static_cast<std::uint64_t>(
+            train_result_to_compute::compute_out_of_bag_error_per_observation)) {
```

Comment:
```suggestion
            train_result_to_compute::compute_out_of_bag_error_per_observation) != 0) {
```

Reply:
the same for other

Reply:
It isn't required here. More over it doesn't make it more informative.